### ISSUE TITLE
[Rust] Mention meaning of asterisk * sign?

### ISSUE THREAD

**Issue by nilsnh** [adambard/learnxinyminutes-docs#1398]
At the end of the explanation there's this code.

``` rust
let mut var2 = 4;
let ref_var2: &mut i32 = &mut var2;
*ref_var2 += 2;

println!("{}", *ref_var2); 
```

It seems that the asterisk is included `*ref_var2 += 2;` because it mutates the variable. But then it seems a bit unclear that the println call is `println!("{}", *ref_var2);` and not `println!("{}", ref_var2);` without the asterisk as the variable has already been mutated? 

In short, the significance of the asterisk seems a bit unclear in the guide.

**Comment by levibostian** [comment]
Mind making a pull request for this? That will help with the discussion and making it better. I don't know Rust so a PR will help.

**Comment by thePsguy** [comment]
Mutation does not transfer the value of the variable to the ref. It just mutably allows it to refer to it.
Also, ref_var2 is defined as a &mut i32, so it can only refer to i32 variables, not store them itself.

**Comment by Razican** [comment]
The asterisk is a dereference of the reference, which means is "make the number referenced by `ref_var2` add 2". That's because ref_var2 is a reference to var2.

**Comment by Oire** [comment]
Fixed via #1463.
